src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -614,18 +615,13 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse) {
 
         // Set correlationId in the result
         if (correlationIdInHeader != null && !correlationIdInHeader.isEmpty()) {
-            try {
-                UUID correlation = UUID.fromString(correlationIdInHeader);
-                if (!correlation.equals(mRequest.getCorrelationId())) {
-                    Logger.w(TAG, "CorrelationId is not matching", "",
-                            ADALError.CORRELATION_ID_NOT_MATCHING_REQUEST_RESPONSE);
-                }
-
-                Logger.v(TAG, "Response correlationId:" + correlationIdInHeader);
-            } catch (Exception ex) {
-                Logger.e(TAG, "Wrong format of the correlation ID:" + correlationIdInHeader, "",
-                        ADALError.CORRELATION_ID_FORMAT, ex);
+            UUID correlation = UUID.fromString(correlationIdInHeader);
```

Comment:
I'm not sure about your concern. The comparison and warning are stayed at their place. I just remove try-catch block since nothing actually throws any excepted exception here